src/components/WorldMap.js

```diff
@@ -48,33 +58,50 @@ export default function WorldMap() {
       });
   }, []);
 
-  let textContent;
-  if (visitedCountries === 0) {
-    textContent = "You didn't select any Country yet";
-  } else if (visitedCountries === 1) {
-    textContent = `You have visited ${visitedCountries} Country in the World`;
-  } else {
-    textContent = `You have visited ${visitedCountries} Countries in the World`;
-  }
+  useEffect(() => {
+    const myMarkers = getMarkersFromLocalStorage();
+    setMarkers(myMarkers);
+    setNumberOfVisitedCountries(getCountriesCountFromLocalStorage());
+  }, []);
 
-  function handleClick(e, latlng) {
+  function handleClick(e, latlng, name) {
     if (e.target.checked === true) {
-      setVisitedCountries(visitedCountries + 1);
-      setMarkers([...markers, latlng]);
+      setNumberOfVisitedCountries(numberOfVisitedCountries + 1);
+      setMarkers([...markers, { latlng, name }]);
+      addMarkerToLocalStorage({ latlng, name });
+      addCountryCountToLocalStorage();
     } else {
-      setVisitedCountries(visitedCountries - 1);
+      setNumberOfVisitedCountries(numberOfVisitedCountries - 1);
       function filterFunction(country) {
-        return country !== latlng;
+        return country.name !== name;
       }
       const filteredMarkers = markers.filter(filterFunction);
```

Comment:
In our opinion this is hard to read. We thought the else body is done in line 77. Maybe you can use an arrow function in the filter method to get the confusion away.